//package ;

message BaseAttrProto
{

  enum ePart
  {
    kPart_Null = 0;
    kPart_Fighter = 1;//战机
    kPart_Armor = 2;//装甲
    kPart_Wingman = 3;//僚机
    kPart_Attache = 4;//附武器
  }

  enum eQuality
  {
    kQuality_Gray = 0;
    kQuality_White = 1;
    kQuality_Green= 2;
    kQuality_Blue = 3;
    kQuality_Purple = 4;
    kQuality_Orange = 5;
  }

  enum eQualityRank
  {
    kQuaRank_0 = 0;
    kQuaRank_1 = 1;
    kQuaRank_2 = 2;
    kQuaRank_Max = 3;
    kQuaRank_Gold = 4;
  }
    required int32 sid = 1[default = 0];
    required int32 eid = 2[default = 0];
    required ePart part = 3[default = kPart_Null];
    required int32 level = 4[default = 1];
    required eQuality quality = 5[default = kQuality_White];
    required eQualityRank quaRank = 6[default = kQuaRank_0];

    optional int32 hp = 7;
    optional int32 defence = 8;
    optional int32 atk = 9;
    optional int32 atkSpeed = 10;
}

message UserInfoRecordProto
{
    required int32 userID = 1[default = 0];
    required string userName = 2[default = "T"];
    required int32 currentFighterID = 3[default= 0];
    repeated string testName = 5;
    repeated BaseAttrProto unitAttr = 4;
}